server/src/server.ts

```diff
@@ -161,30 +152,31 @@ connection.listen()
 documents.onDidOpen((event) => {
   const textDocument = event.document
   if (textDocument.getText().length > 0) {
-    documentAsTextMap.set(textDocument.uri, textDocument.getText().split(/\r?\n/g))
+    analyzer.documentAsTextMap.set(textDocument.uri, textDocument.getText().split(/\r?\n/g))
   }
 
   setSymbolScanner(new SymbolScanner(textDocument.uri, contextHandler.definitionProvider))
 })
 
 documents.onDidChangeContent(async (event) => {
   const textDocument = event.document
-  documentAsTextMap.set(textDocument.uri, textDocument.getText().split(/\r?\n/g))
+  analyzer.documentAsTextMap.set(textDocument.uri, textDocument.getText().split(/\r?\n/g))

```

Comment:
what about we replace this with a method into analyzer in order to avoid repetition?
``` typescript
public addDocumentAsText(textDocument: TextDocument): void {
  this.documentAsTextMap.set(textDocument.uri, textDocument.getText().split(/\r?\n/g))
}
```

or something

Reply:
Analyzer's `uriToAnalyzedDocument` already has a TextDocument object, so I use it to replace the `documentAsTextMap`